### PR TITLE
Zadanie 3: Illia (Hanzoze)

### PR DESCRIPTION
**Pull request by Hanzoze**

### CODE REVIEW

**Review by MrChazar — APPROVED** [review]
Uwagi:
- dobrze opisana praca
- poprawne wykorzystanie warstw konwolucjynych i dropoutów
Praca na plus

**Review by ozio1212 — APPROVED** [review]
Mega fajna praca, bardzo przejrzysta i przyjemna w czytaniu. Wyniki i wnioski też mega spoczko 👍

**Review by rozamazurek — COMMENTED** [review]
Bardzo fajna praca, przyjemnie się to czytało, dużo wniosków, wszystko ładnie opisane, przez co wiadomo co się dzieje i co jest testowane. Fajna analiza wpływu różnych parametrów na model. Nie ma się do czego przyczepić.

**Comment by GregW04** [comment]
@Hanzoze  Cześć przychodzę z code review:

- ustawianie stałego ziarna +
- podział na train, val, split +
- wygodne do czytania nadawanie zmiennych +
- wygodne do czytania nadawanie funkcji (FlexibleCNN) + 
- fajne krok po kroku rozszerzanie modelu z celnymi obserwcjami i wnioskami
- BatchNormalization +
- Wydłużenie sieci + 
- super intuicyjne i celne wnioski + 
- Ogółem badania przeprowadzone w czytelny sposób, widać logiczny ciąg eksperymentowania poparty krótkimi na  temat komentarzami.
- brakuje innych metryk jak np. recall, precision i F1-score oraz ROC-AUC -
-  Co bym dodał:
-  Barploty zestawiające metryke i loss różnych modeli (dla konkretnych (1/ostatnia) epok lub uśrednione)
- Eksperymenty z różnymi optimizerami oraz ich learning-rate'ami
- Eksperymenty z różnymi funkcjami aktywacji 
- Eksperymenty z BATCH_SIZE
- Spróbować jakiejś gotowej insfrastruktury (jak czujesz się na siłach np. z huggingface)
- Ogółem bardzo konkretna i fajna w czytaniu praca, dzięki
